packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -108,6 +106,19 @@ export const createGiveaway = async (req: Request, res: Response) => {
     });
     giveawayId = giveaway._id;
 
+    await GiveawayLog.create({
```

Comment:
Have you tried something like this: https://www.npmjs.com/package/mongoose-history-plugin There are a couple of plugins that log changes but I've never used them so I'm not sure they work

Reply:
Tested this solution and found this out:
https://github.com/Masquerade-Circus/mongoose-history-plugin/issues/28

It seems that there is no ts version available, and the app crashes due to not recognizing the history object, and i couldn't find any relevant solution to this, so this seems to be a blocker.

I have also tried using middleware hooks: https://mongoosejs.com/docs/middleware.html and its seems to be a more elegant and sturdy solution, however we would still be handling database interaction in the same way, we would just be passing log creation calls to a pre hook on each operation, as well as losing some information (such as user info).

I am now currently trying out mongoose change streams: https://mongoosejs.com/docs/change-streams.html though it seems to be a similar solution to the one above.